### ISSUE TITLE
iBook G4 Mouse not showing

### ISSUE THREAD

**Issue by ojt1998** [Wack0/maciNTosh#42]
Hi there,

I’ve used the latest released iso to install NT4 on an iBook G4. I’ve followed the setup steps and selected no mouse but unfortunately it doesn’t seem to have then reactivated the mouse at the gui part of setup or once on the desktop.

Have I missed something?

**Comment by Wack0** [comment]
Which iBook G4 is this (PowerBookx,x)?

Can I see the Open Firmware device tree?

**Comment by ojt1998** [comment]
It’s the Powerbook6,7

![image](https://github.com/user-attachments/assets/9bb7a4ca-6555-48e6-a9af-bb6102ebd34e)

**Comment by Wack0** [comment]
ok, can I see the full device tree? with just devalias output I can't be 100% sure.

under open firmware:

```
dev /
ls
```

**Comment by ojt1998** [comment]
![IMG_9714](https://github.com/user-attachments/assets/5de65955-5ab3-45db-90d9-dfaf7941386c)
![IMG_9715](https://github.com/user-attachments/assets/ce21ddd5-1a8b-4a41-a0c3-8d65aa6b9c97)

**Comment by Wack0** [comment]
interesting, seems there's two USB mice present (one on a hub with a keyboard), yet ADB keyboard is present. (as well as ADB mouse)

evidently of those, ADB keyboard works but not mouse.

Interesting that ADB keyboard works even though USB keyboard is present. I'm going to say mouse won't work on this system until USB support happens.

**Comment by ojt1998** [comment]
No worries - just my luck 😂

I tried booting on my Powerbook G4 but it doesn’t load into firmware and gets stuck on the initial step…

I shall keep waiting 😅

**Comment by Wack0** [comment]
that would be a different issue. where does it get stuck? on a black screen? and what PowerBookx,x is that?

**Comment by Wack0** [comment]
More relevant to this issue, I've used NT without a mouse quite a bit (never did get working mouse on my Wii port for instance), and NT is somewhat usable without a mouse, just annoying to get around with just keyboard shortcuts and CLI, etc.

**Comment by ojt1998** [comment]
It’s a PowerBook 5,6

Looking at the device tree there isn’t adb so that is likely the cause of the hang. I did assume that earlier as the hang is at “Init adb…”

**Comment by ojt1998** [comment]
> More relevant to this issue, I've used NT without a mouse quite a bit (never did get working mouse on my Wii port for instance), and NT is somewhat usable without a mouse, just annoying to get around with just keyboard shortcuts and CLI, etc.

Yeah I managed to get office 97 installed but like you say it’s a bit of a pain!

**Comment by Wack0** [comment]
> Looking at the device tree there isn’t adb so that is likely the cause of the hang. I did assume that earlier as the hang is at “Init adb…”

that one's probably waiting forever polling PMU interrupt status looking for an ADB interrupt that never arrives. maybe I should add a timeout or something to that, although without working USB support it's not like that'll do much.

**Comment by ojt1998** [comment]
I’ve done some digging on the iBook and it looks as though the trackpad is on adb but the usb mouse has a higher number - could that be why NT doesn’t detect it?

**Comment by Wack0** [comment]
nope, the higher ID for mouse is intended and expected. the Open Firmware implementation on these systems when creating ADB device in the device tree does blindly create entries for keyboard and mouse even if one or other doesn't really exist.

**Comment by WamWooWam** [comment]
The late model iBook G4's (that's models A1133 and A1134) have multi-touch trackpads which are USB, therefore will not be supported by NT at this time
